codec-http2/src/test/java/io/netty/handler/codec/http2/CleartextHttp2ServerUpgradeHandlerTest.java

```diff
@@ -175,6 +183,48 @@ public void downgrade() throws Exception {
         assertNull(channel.readInbound());
     }
 
+    @Test
+    public void usedHttp2Codec() throws Exception {
+        final Http2Codec http2Codec = new Http2CodecBuilder(true, new ChannelInitializer<Channel>() {
+            @Override
+            protected void initChannel(Channel ch) throws Exception {
+            }
+        }).build();
+        UpgradeCodecFactory upgradeCodecFactory = new UpgradeCodecFactory() {
+            @Override
+            public UpgradeCodec newUpgradeCodec(CharSequence protocol) {
+                return new Http2ServerUpgradeCodec(http2Codec);
+            }
+        };
+        http2ConnectionHandler = http2Codec.frameCodec().connectionHandler();
+
+        userEvents = new ArrayList<Object>();
+
+        HttpServerCodec httpServerCodec = new HttpServerCodec();
+        HttpServerUpgradeHandler upgradeHandler = new HttpServerUpgradeHandler(httpServerCodec, upgradeCodecFactory);
+
+        CleartextHttp2ServerUpgradeHandler handler = new CleartextHttp2ServerUpgradeHandler(
+                httpServerCodec, upgradeHandler, http2Codec);
+        channel = new EmbeddedChannel(handler, new ChannelInboundHandlerAdapter() {
+            @Override
+            public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
+                userEvents.add(evt);
+            }
+        });
+
+        channel.writeInbound(Http2CodecUtil.connectionPrefaceBuf());
```

Comment:
Addressed by 7e4fa2dfdd832d00a0ad76d7ea4ee8533791ad7c

Please let me know if the pr is ready for merge, I can squash the commits.